# My Datascience Project
This project will assess the number of users who use an app. This data will likely show developers what type of apps are likely to attract more users. 

In [1]:
from csv import reader 
#AppleStore.csv
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
apple = list(read_file)
apple_header = apple[0] 
apple = apple[1:]

#Googleplaystore.csv
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
google = list(read_file)
google_header = google[0]
google = google[1:]

print(apple_header)
print(len(apple_header))
print(google_header)
print(len(google_header))

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
16
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
13



## Exploring two data sets


In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') #adds a new (empty) line after each row
        
    if rows_and_columns:
        print('Number of rows:',len(dataset))
        print('Number of columns:',len(dataset[0]))
        
print(google_header)
print('\n')
explore_data(google, 0, 3, True)
## useful categories: app, catgegory, rating, reviews, installs, type, price, content rating 
## and genres
        

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


## apple data set

In [3]:
print(apple_header)
print('\n')
explore_data(apple, 0, 3, True)
## track_name, currency, price, rating_count_ver, user_rating 
## columns are not labeled the same as googlestore

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


## Data cleaning
apps that are free and are English-speaking audience 
print row according to discussion on kaggle and compare it to header and another row with all data values

In [4]:
print(google[10472]) #Missing category and genre
print('\n')
print(google_header) #header checking for missing data in 10472
print('\n')
print(google[0]) #correct row with all data 
print('\n')

## in row 10742 can a catgeory be a float? 
## if category can be a float then rating is incorrect
## rating is 19 out of possible 5 

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']




In [5]:
print(len(google))
del google[10472]##I ran more than once and had to restart kernel
print(len(google))

10841
10840


## Deleting duplicate entries 
- We don't want to count certain apps more than one when we analyze data. 
- Remove duplicates and keep only one entry 
- Remove the older data sets and keep the most recent

In [6]:
for app in google: 
    name = app[0]
    if name == 'Instagram': 
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


- create two lists 
- loop through google play data set and for each interations:
- save the app name to a variable named **name**
- if **name** in **unique_apps** list append **name** to the **duplicate_apps** list
- else(if **name** isn't in **unique_apps** append name to **unique_apps** list

In [7]:
duplicate_apps = []
unique_apps = []

for app in google:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else: 
        unique_apps.append(name)
        
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


* We don't want to remove the duplicates randomly
* We need to remove the duplicate entries and keep only one entry per app
* Look at the apps number of reviews. The reviews indicate that the data was collected at different times 
* The reviews with the highest number of reviews is the most recent 


In [9]:
reviews_max = {}

for app in google:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
print(len(reviews_max)) ##9659 entries  

ValueError: could not convert string to float: '3.0M'

In [9]:
print('Expected length:', len(google) - 1181)

Expected length: 9659


* create two empty lists google_clean and already_added
* loop through google data set, don't include header row
* isolate the name of the app and the number of reviews 
* add current row (app) to the google_clean list and the app name (name) to the already_added list 
* if the number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary AND 
* the name of the app is not already in the already_added list. We add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same or more than one entry. 

In [10]:
google_clean = []
already_added = []

for app in google:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        google_clean.append(app)
        already_added.append(name) 

In [11]:
explore_data(google_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


In [12]:
print(apple[813][1])
print(apple[6731][1])

print(google_clean[4412][0])
print(google_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


In [14]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


In [15]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


* the two apps above are English speaking apps 
* incorrectly identified because of the emoji and character use ™ 
* these fall outside the ASCII range and have corresponding numbers over 127 
* to prevent data loss 
* remove an app if its name contains more than 3 non ASCII characters


In [16]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
    

True
True


In [17]:
google_english = []
apple_english = []

for app in google_clean:
    name = app[0]
    if is_english(name):
        google_english.append(app)
        
for app in apple:
    name = app[1]
    if is_english(name):
        apple_english.append(app)
        
explore_data(google_english, 0, 3, True)
print('\n')
explore_data(apple_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

In [18]:
google_final = []
apple_final = []

for app in google_english: 
    price = app[7]
    if price == '0':
        google_final.append(app)
        
for app in apple_english:
    price = app[4]
    if price == '0.0':
        apple_final.append(app)
        
print(len(google_final))
print(len(apple_final))

8864
3222


## Most Common Genres for Each Market
We need to find apps that are successful in both markers, so that we can add our app to the apple and g oogle store.

* build a frequency table for the prime_genre column in Apple Store
* build a frequency table for genres and category columns in google play store

In [33]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [34]:
display_table(apple_final, -5) ##prime genre -5 (5th to last element)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


## Notes 
* most common genre = Games 58.16%
* Entertainment = 7.8 % 
* photo and video = 4.9% 
* education = 3.6% 
* social networking = 3.2%


In [35]:
display_table(google_final, 1)## category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

## Most common genres for google app store
* categories are listed differently in the google store 
* it appears that the google store as more application and practicality than the apple store. 

In [37]:
display_table(google_final, -4) ##genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

##Notes between categories and genre 
* genres column produces more categories than 
categories list 
* apple store has more apps designed for fun that are being downloaded where as google play as a balanced amount for practical and fun apps

In [41]:
genres_apple = freq_table(apple_final, -5)

for genre in genres_apple:
    total = 0
    len_genre = 0
    for app in apple_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Navigation apps have the highest number of user reviews 

In [43]:
for app in apple_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [44]:
for app in apple_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


## app profile recommendation for the app store
* Shopping apps which include clothing, shoes, buying goods
* Business apps which include venmo, paypal, cash app, banking

In [45]:
display_table(google_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


## Notes 
* We don't know whethere an app with 100,000+ installs has, 200,000, or 350,000.
* We don't need precise data for our purposes 
* We only want to find out which app genres attract most users 
* Consider that an app with 100,000 + installs has 100,000 installs
* An app with 1,000,000+ installs has 1,000,000 installs and so on.


In [46]:
categories_google = freq_table(google_final, 1)

for category in categories_google:
    total = 0
    len_category = 0
    for app in google_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

Communication apps have the most installs at 38,456,119 
* We will take a look at the apps that have the most installs within communication apps

In [47]:
for app in google_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

remove communications apps that have over 100 million installs

* Communication installs 38,456,119
* Removed communication 36,034,85

There are a small amount of popular apps that skew the average

In [53]:
under_100_m = []

for app in google_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

In [55]:
for app in google_final:
    if app[1] == 'SHOPPING':
        print(app[0], ':', app[5])

Amazon for Tablets : 10,000,000+
OfferUp - Buy. Sell. Offer Up : 10,000,000+
Shopee - No. 1 Online Shopping : 10,000,000+
Shopee: No.1 Online Shopping : 10,000,000+
Kroger : 5,000,000+
Walmart : 10,000,000+
eBay: Buy & Sell this Summer - Discover Deals Now! : 100,000,000+
letgo: Buy & Sell Used Stuff, Cars & Real Estate : 50,000,000+
Amazon Shopping : 100,000,000+
Lazada - Online Shopping & Deals : 50,000,000+
OLX - Buy and Sell : 50,000,000+
The wall : 1,000,000+
Flipp - Weekly Shopping : 10,000,000+
Shrimp skin shopping: spend less, buy better : 5,000,000+
Lotte Home Shopping LOTTE Homeshopping : 5,000,000+
Horn, free country requirements : 1,000,000+
Jiji.ng : 1,000,000+
GS SHOP : 10,000,000+
The birth : 50,000,000+
Home & Shopping - Only in apps. 10% off + 10% off : 10,000,000+
EHS Dongsen Shopping : 1,000,000+
bigbasket - online grocery : 5,000,000+
Bukalapak - Buy and Sell Online : 10,000,000+
Life market : 1,000,000+
Jabong Online Shopping App : 10,000,000+
Family Dollar : 1,000

In [56]:

for app in google_final:
    if app[1] == 'SHOPPING' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

eBay: Buy & Sell this Summer - Discover Deals Now! : 100,000,000+
Amazon Shopping : 100,000,000+
Flipkart Online Shopping App : 100,000,000+
AliExpress - Smarter Shopping, Better Living : 100,000,000+
Wish - Shopping Made Fun : 100,000,000+


There are not as many apps in the 100,000,000+ range as much as there are in the 1,000,000+ - 50,000,000+ range. 
* Consider that these apps are free but the purchases of items on these apps may not be free. 
* less installs because online shopping requires money

In [57]:
for app in google_final:
    if app[1] == 'SHOPPING' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Amazon for Tablets : 10,000,000+
OfferUp - Buy. Sell. Offer Up : 10,000,000+
Shopee - No. 1 Online Shopping : 10,000,000+
Shopee: No.1 Online Shopping : 10,000,000+
Kroger : 5,000,000+
Walmart : 10,000,000+
letgo: Buy & Sell Used Stuff, Cars & Real Estate : 50,000,000+
Lazada - Online Shopping & Deals : 50,000,000+
OLX - Buy and Sell : 50,000,000+
The wall : 1,000,000+
Flipp - Weekly Shopping : 10,000,000+
Shrimp skin shopping: spend less, buy better : 5,000,000+
Lotte Home Shopping LOTTE Homeshopping : 5,000,000+
Horn, free country requirements : 1,000,000+
Jiji.ng : 1,000,000+
GS SHOP : 10,000,000+
The birth : 50,000,000+
Home & Shopping - Only in apps. 10% off + 10% off : 10,000,000+
EHS Dongsen Shopping : 1,000,000+
bigbasket - online grocery : 5,000,000+
Bukalapak - Buy and Sell Online : 10,000,000+
Life market : 1,000,000+
Jabong Online Shopping App : 10,000,000+
Family Dollar : 1,000,000+
Jumia online shopping : 10,000,000+
Mercado Libre: Find your favorite brands : 50,000,000+


The popular item for SHOPPING seems to be most popular in buying and exchanging goods within the 1,000,000+ - 50,000,000+ install range. This might be because the amount of money that it takes to buy items online versus apps that are free to download and have free music, books, communication, or navigation. 

* Coupon apps and apps known for better discounts are more installed consider shopping apps that allow more discounts and coupons.  
* shopping apps with more practicality for home use and lifestyle appear to have more installs like Amazon, home depot, wayfair, ebay. 
* Suggestions for client would be to develop an app that is a blend of the highest downloaded app (communication apps) and least downloaded app (shopping apps). 
* Shopping apps have less downloads and have the potential to be sucessful when paired with communication apps. 
* An app developed specificially for the shopping and buying experience of people of color would allow people of color/diversity to contact other buyers who have given a product review. 
* This contact would give potential buyers access to former buyers social media to see products in social media feed and ability to live communicate through messenger feature. 
* This app suggestion comes from my own experience as a person of multiple culture backgrounds I am more likely to purchase an beauty products, clothing, and home goods from people who are most similiar to my ethnicity, culture, race, or lived experiences. 

## Conclusions 
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets. 

We concluded that taking shopping apps could be more profitable for both the Google Play and the App Store markets if the client develops an app that is focused on the shopping and buying experience through the lens of a person of color. This app would give access to social media accounts and messenger so that the buyer and reviewer can communicate about products purchased and used. 
